packages/dev/core/src/FlowGraph/flowGraphEventCoordinator.ts

```diff
@@ -0,0 +1,28 @@
+import { Observable } from "../Misc/observable";
+import type { FlowGraphCustomEvent } from "./flowGraphCustomEvent";
+
+/**
+ * This class centralizes the handling of events in the flow graph.
+ * If a block fires an event, it will be handled by this class.
+ * It will also notify any blocks that are listening to that event.
+ * @experimental
+ */
+export class FlowGraphEventCoordinator {
+    private _customEventsMap: Map<string, Observable<FlowGraphCustomEvent>> = new Map();
+
+    getCustomEventObservable(id: string): Observable<FlowGraphCustomEvent> {
+        let observable = this._customEventsMap.get(id);
+        if (!observable) {
+            observable = new Observable<FlowGraphCustomEvent>();
+            this._customEventsMap.set(id, observable);
+        }
+        return observable;
+    }
+
+    notifyCustomEvent(id: string, data: any) {
+        const observable = this._customEventsMap.get(id);
+        if (observable) {
+            observable.notifyObservers({ id: id, data });

```

Comment:
Is this happening frequently ? if yes, it might be nice to think of a way not requiring instantiation.

Reply:
i don't think this happens a lot, but of course Carol knows best :-)

What I can say is, that if you already use `data` , why not use ` id`  alone, without the `: id`.

I think it is fine for custom events.

Reply:
It can happen frequently if there's a ton of events being used... maybe we could use positional (id, data) parameters in this case then?

Reply:
Oh, dang, I forgot notifyObservers only takes one argument... through, maybe we could notify only the event data? We might not need the id anyways.